packages/client/components/PokerEstimateHeaderCard.tsx

```diff
@@ -23,26 +69,21 @@ const PokerEstimateHeaderCard = (props: Props) => {
     return <PokerEstimateHeaderCardParabol task={task} />
   }
   // it's an integrated task, but the service might be down
-  if (!integration) {
+  const headerFields = getHeaderFields(integration)
+  if (!headerFields) {
     return <PokerEstimateHeaderCardError service={'Integration'} />
   }
 
-  if (integration.__typename === 'JiraIssue' || integration.__typename === 'JiraServerIssue') {
-    const name = integration.__typename === 'JiraIssue' ? 'Jira' : 'Jira Server'
-    return (
-      <PokerEstimateHeaderCardContent
-        summary={integration.summary}
-        descriptionHTML={integration.descriptionHTML}
-        url={integration.jiraUrl}
-        linkTitle={`${name} Issue #${integration.issueKey}`}
-        linkText={integration.issueKey}
-      />
-    )
-  }
-  if (integration.__typename === '_xGitHubIssue') {
-    return <PokerEstimateHeaderCardGitHub issueRef={integration} />
-  }
-  return null
+  const {cardTitle, descriptionHTML, url, linkTitle, linkText} = headerFields
+  return (
+    <PokerEstimateHeaderCardContent
+      cardTitle={cardTitle}
+      descriptionHTML={descriptionHTML}
+      url={url}
+      linkTitle={linkTitle}
+      linkText={linkText}
```

Comment:
+1 Especially if you move the `IntegrationHeaderFields` type over to `PokerEstimateHeaderCardContent` and rename it to `PokerEstimateHeaderCardContentProps` you could just spread them here
```
<PokerEstimateHeaderCardContent {...headerFields}/>
```

Reply:
That's much cleaner. Update pushed!